src/Internal/QueryM.purs

```diff
@@ -405,159 +291,6 @@ instance Parallel (QueryMExtended r ParAff) (QueryMExtended r Aff) where
   sequential :: QueryMExtended r ParAff ~> QueryMExtended r Aff
```

Comment:
Looks like:

```
type QueryEnv (r :: Row Type) =
  { config :: QueryConfig
  , runtime :: QueryRuntime
  , extraConfig :: { | r }
  }
  ```
  
  is the only place where `extraConfig` got left, and it is not used.
  
  Also `QueryMExtended` is not used anymore.
  
  Can we remove them all?

Reply:
Not exactly, `QueryM` and `ParQueryM` are aliases of `QueryMExtended`, however we can remove `r` and `extraConfig`. I also renamed `QueryMExtended` to `QueryMT` to indicate it is a monad transformer of some base monad; `Aff` for `QueryM` and `ParAff` for `ParQueryM`.

Reply:
We could make `QueryM` and `ParQueryM` `data` instead of `type`, but none of this faces the user and it's not going to save us much

Reply:
`ParQueryM` is not used too, as far as I see.

Reply:
> But ParQueryM is not used too, as far as I see.

Good point

Reply:
> Good point

Nevermind, `ParQueryM` is used indirectly at https://github.com/Plutonomicon/cardano-transaction-lib/blob/cad5b9c7341c33d4df6e3a926148cf4cd38d03b0/src/Internal/QueryM/Kupo.purs#L302

We could still remove `ParQueryM` and unwrap `QueryM` at that point to use `Aff`'s `parTraverse`, and then wrap again, but it doesn't really matter